cli/internal/cmd/apply.go

```diff
@@ -858,3 +859,9 @@ type imageFetcher interface {
 		image, region string, useMarketplaceImage bool,
 	) (string, error)
 }
+
+func logDebug(logger debugLog, filehandler file.Handler, msg string) {
+	logger.Debug(msg)
+
+	_ = filehandler.Write(constants.InitDebugLogFile, []byte(msg+"\n"), file.OptAppend)
+}
```

Comment:
Why not create a new debugFileLogger struct which satisfies the `debugLog` interface and writes messages to a file?
That would easily ensure all messages are logged correctly, and also makes it more easily usable for other commands.
e.g. :
```Go
type debugFileLogger struct {
	fileHandler file.Handler
	log         debugLog
}

func (l *debugFileLogger) Debug(msg string, args ...any) {
	l.log.Debug(msg, args...)
	_ = l.fileHandler.Write("file", []byte(msg+"\n"), file.OptAppend)
}
```

Reply:
Yes, that's a good idea, I should've thought of that myself.